### PR TITLE
Added node id to the change event

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: chitchat/src/lib.rs [hunk 1]
```diff
@@ -282,7 +282,8 @@ impl Chitchat {
         self.cluster_state.compute_digest(dead_nodes)
     }
 
-    /// Subscribe a callback that will be called every time a key matching the  supplied prefix.
+    /// Subscribe a callback that will be called every time a key matching the supplied prefix
+    /// is inserted or is updated.
```
Comment (guilload) [inline chitchat/src/lib.rs:286]:
```suggestion
    /// Subscribes a callback that will be called every time a key matching the supplied prefix
    /// is inserted or updated.
```

File: chitchat/src/lib.rs [hunk 2]
```diff
@@ -291,20 +292,40 @@ impl Chitchat {
...
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct KeyChangeEvent<'a> {
+    key: &'a str,
+    value: &'a str,
+    node: &'a ChitchatId,
+}
+
+impl<'a> KeyChangeEvent<'a> {
+    pub fn strip_key_prefix(&self, prefix: &str) -> Option<KeyChangeEvent> {
```
Comment (guilload) [inline chitchat/src/lib.rs:319]: Does it need to be public?

**Review by guilload — APPROVED** [review]

File: chitchat/src/lib.rs (hunk 2 above)
Comment (fulmicoton) [reply chitchat/src/lib.rs]: no. removing pub

File: chitchat/src/lib.rs (hunk 1 above)
Comment (fulmicoton) [reply chitchat/src/lib.rs]: manually edited the text.

**Pull request merged**